cognite/neat/_data_model/deployer/_deployer.py

```diff
@@ -0,0 +1,147 @@
+from dataclasses import dataclass
+from datetime import datetime, timezone
+from typing import Literal, cast
+
+from cognite.neat._client import NeatClient
+from cognite.neat._data_model.models.dms import RequestSchema, T_DataModelResource, T_ResourceId
+
+from ._differ import ItemDiffer
+from ._differ_container import ContainerDiffer
+from ._differ_data_model import DataModelDiffer
+from ._differ_space import SpaceDiffer
+from ._differ_view import ViewDiffer
+from .data_classes import (
+    AppliedChanges,
+    DataModelEndpoint,
+    DeploymentResult,
+    ResourceChange,
+    ResourceDeploymentPlan,
+    SchemaSnapshot,
+    SeverityType,
+)
+
+
+@dataclass
+class DeploymentOptions:
+    """Configuration options for deployment.
+
+    Attributes:
+        dry_run (bool): If True, the deployment will be simulated without applying changes. Defaults to True.
+        auto_rollback (bool): If True, automatically roll back changes if deployment fails. Defaults to True.
+        max_severity (SeverityType): Maximum allowed severity of changes to proceed with deployment.
+            Defaults to SeverityType.SAFE.
+        modus_operandi (Literal["partial", "overwrite"]): Deployment mode. If "partial", only add/modify resources
+            specified in the data model. If "overwrite", remove resources not present in the data model.
+            Defaults to "partial".
+    """
+
+    dry_run: bool = True
+    auto_rollback: bool = True
+    max_severity: SeverityType = SeverityType.SAFE
+    modus_operandi: Literal["partial", "overwrite"] = "partial"
+
+
+class SchemaDeployer:
+    def __init__(self, client: NeatClient, options: DeploymentOptions | None = None) -> None:
+        self.client: NeatClient = client
+        self.options: DeploymentOptions = options or DeploymentOptions()
+        self._results: DeploymentResult | None = None
+
+    @property
+    def results(self) -> DeploymentResult:
+        if self._results is None:
+            raise RuntimeError("SchemaDeployer has not been run yet.")
+        return self._results
+
+    def run(self, data_model: RequestSchema) -> None:
+        self._results = self.deploy(data_model)
```

Comment:
Prepare for `OnSuccess` will add that as a separate PR as I have a suggestion for how to do it.